### ISSUE TITLE
Add RBAC policy -> namespace allowance in the KubernetesExecutor

### ISSUE THREAD

**Issue by dimberman** [apache/airflow#10836]
**Description**

**Use case / motivation**

**Related Issues**

**Comment by maganaluis** [comment]
# Description

Add RBAC policy to ensure Roles can only execute DAGs against an assigned namespaces for Kubernetes Executor.

# Use Case / Motivation

If we have role called "ds-project-xyz" and we assign users to this project, right now users can use this role to give their DAGs access control, this will ensure that other users that are not part of this project don't have access to this DAG. 

However we also want to make sure that the tasks for this DAG run on the assigned namespace for this project "xyz" as they could have potentially sensitive data in that namespace that we don't want to get mixed up with other projects, or they also have access to sensitive credentials via Secrets. In either case we need a way for Roles to have some sort of connection to a specific namespace. 

```
with DAG(DAG_NAME,
     schedule_interval='@daily',
     default_args=default_args,
     access_control={
         'ds-project-xyz': {'can_dag_read'},
     },
     ) as dag:
DummyOperator(task_id='run_this_1') >> DummyOperator(
    task_id='run_this_2') >> DummyOperator(task_id='run_this_3')
```

**Comment by maganaluis** [comment]
@dimberman Feel free to assign me to this, also feel free to point me to any resources. I haven't had a chance to develop against 2.0 so any supporting documentation would be helpful.

**Comment by dimberman** [comment]
@maganaluis assigned!

**Comment by eladkal** [comment]
@maganaluis are you still working on this feature?

**Comment by maganaluis** [comment]
@eladkal No, we switched to Kubeflow. I don't have the bandwidth to support both. Could you remove me from this task?

**Comment by bdsoha** [comment]
@eladkal If I am not mistaken, this has been implemented. Can this be closed?
